### PR TITLE
Support mtry.ratio and sampsize.ratio

### PR DESCRIPTION
**Pull request by mllg**
To be able to tune conveniently...

### CODE REVIEW

File: R/helpers.R [hunk 1]
```diff
@@ -57,3 +57,27 @@ pprob_to_matrix <- function(pp, task) {
   colnames(y) <- task$class_names
   y
 }
+
+convert_ratio = function(pv, target, ratio, n) {
```
Comment (RaphaelS1) [inline R/helpers.R:61]: I'm struggling to see how this works, can you please add a one-line comment explaining what's happening and in one of the learners add a test to show it works. From what I can tell the following should fail (but doesn't):

```R
convert_ratio(list(mtry = 2, mtry.ratio = 0.2), "mtry", "mtry.ratio", 5)
```

File: R/helpers.R [hunk 2]
```diff
@@ -57,3 +57,27 @@ pprob_to_matrix <- function(pp, task) {
...
+  to_decimal = function(bits) {
```
Comment (RaphaelS1) [inline R/helpers.R:62]: Does this mean `bits` in the computer sense? If not maybe rename?

File: R/learner_obliqueRSF_surv_obliqueRSF.R [hunk 3]
```diff
@@ -42,6 +43,7 @@ LearnerSurvObliqueRSF = R6Class("LearnerSurvObliqueRSF",
           max_pval_to_split_node = p_dbl(lower = 0, upper = 1, default = 0.5,
             tags = "train"),
           mtry = p_int(lower = 1, tags = "train"),
+          mtry_ratio = p_dbl(0, 1, tags = "train"),
```
Comment (RaphaelS1) [inline R/learner_obliqueRSF_surv_obliqueRSF.R:46]: Is there a reason why sometimes it's `mtry_ratio` and others `mtry.ratio` ?

File: R/learner_obliqueRSF_surv_obliqueRSF.R [hunk 4]
```diff
@@ -76,11 +78,12 @@ LearnerSurvObliqueRSF = R6Class("LearnerSurvObliqueRSF",
   private = list(
     .train = function(task) {
       pv = self$param_set$get_values(tags = "train")
+      pv = convert_ratio(pv, "mtry", "mtry_ratio", length(task$feature_names))
       targets = task$target_names
 
       mlr3misc::invoke(
         obliqueRSF::ORSF,
-        data     = as.data.frame(task$data()),
+        data     = data.table::setDF(task$data()),
```
Comment (RaphaelS1) [inline R/learner_obliqueRSF_surv_obliqueRSF.R:86]: thanks

File: R/learner_partykit_regr_cforest.R [hunk 5]
```diff
@@ -37,6 +31,7 @@ LearnerRegrCForest = R6Class("LearnerRegrCForest",
           tags = "train"),
         mtry = p_int(lower = 0L, special_vals = list(Inf),
           tags = "train"), # default actually "ceiling(sqrt(nvar))"
+        mtryratio = p_dbl(lower = 0, upper = 1, tags = "train"),
```
Comment (RaphaelS1) [inline R/learner_partykit_regr_cforest.R:34]: As above why here `mtryratio` not `mtry.ratio` or `mtry_ratio` ?

**Review by RaphaelS1 — CHANGES_REQUESTED** [review]
Thanks, a few minor requests for testing.
Can you please also bump patch numbers and add a note to the news about new params

**Comment by mllg** [comment]
> Thanks, a few minor requests for testing.
> Can you please also bump version patch number and add a note to the news about new params

Can do, but this leads inevadibly to to conflicts, right?

File: R/helpers.R (hunk 2 above)
Comment (mllg) [reply R/helpers.R:62]: yes, bits as in computers. this function is now in mlr3misc.

**Comment by mllg** [comment]
Regarding the naming of `mtry.ratio`/`mtryratio`/`mtry_ratio` ... I followed the naming scheme of the package, maybe this is a bad idea?

**Comment by RaphaelS1** [comment]
> Regarding the naming of `mtry.ratio`/`mtryratio`/`mtry_ratio` ... I followed the naming scheme of the package, maybe this is a bad idea?

Yeah this is a weird one, I see your point... Okay maybe it's just strange in review but within learners will look correct

File: R/helpers.R (hunk 2 above)
Comment (RaphaelS1) [reply R/helpers.R]: Okay so should we remove it here then?

**Comment by RaphaelS1** [comment]
> > Thanks, a few minor requests for testing.
> > Can you please also bump version patch number and add a note to the news about new params
> 
> Can do, but this leads inevadibly to to conflicts, right?

Seems to be working fine so far. Because we automate releases in mlr3extralearners it's just useful to keep a clear history of updates within each release.

**Comment by RaphaelS1** [comment]
@mllg  one minor point about the function in mlr3misc above (https://github.com/mlr-org/mlr3extralearners/pull/109#discussion_r704758523) otherwise happy to merge

**Comment by mllg** [comment]
Has already been removed in 40d01a45d166773f8bbe926706eb4b8317035013, GitHub displays outdated code snippets if there are unresolved comments.

**Pull request merged**